### PR TITLE
Auth UI v2 - Unmask otp last digit when focused

### PR DESCRIPTION
**Pull request by IniZio**
ref #3735

### CODE REVIEW

**Comment by tung2744** [comment]
Seems format check failed. Maybe you can install the vscode prettier plugin?

File: authui/src/otpInput.ts [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import { Controller } from "@hotwired/stimulus";
+import { TextFieldController } from './authflowv2/text-field';
```
Comment (louischan-oursky) [inline authui/src/otpInput.ts]: Can we apply `text-field` controller in `__otp_input.html`?

File: authui/src/otpInput.ts [hunk 2]
```diff
@@ -9,6 +10,14 @@ export class OtpInputController extends Controller {
...
   spans: HTMLElement[] = [];
 
+  get textField(): TextFieldController | null {
+    const ctr = this.application.getControllerForElementAndIdentifier(
+      this.inputTarget,
+      "text-field"
+    );
+    return ctr as TextFieldController | null;
+  }
+
```
Comment (louischan-oursky) [inline authui/src/otpInput.ts:20]: I think this is unnecessary. If text-field is applied correctly, it should do its job correctly. If not, there should be a bug we need to fix.

File: authui/src/otpInput.ts [hunk 3]
```diff
@@ -64,6 +73,7 @@ export class OtpInputController extends Controller {
       this.submitTarget.click();
     }
 
+    this.textField?.onInput();
```
Comment (louischan-oursky) [inline authui/src/otpInput.ts:76]: This is unnecessary.

File: authui/src/otpInput.ts (hunk 1 above)
Comment (IniZio) [reply authui/src/otpInput.ts]: The issue is the hidden input is not the one with error state. Maybe I add an input container target.

**Pull request merged**